### ISSUE TITLE
addon/closebrackets.js can not work in mobile

### ISSUE THREAD

**Issue by ahong-qin** [codemirror/CodeMirror#6372]
addon/closebrackets.js can not work in mobile
addon/closetag.js can not work in mobile

**Comment by marijnh** [comment]
It'd be nice if you could put a bit more effort into a bug report. Does 'mobile' mean Chrome Android in this case, or did you test more browsers?

**Comment by ahong-qin** [comment]
addon/closebrackets.js and&nbsp;addon/closetag.js can work in iOS system, including Safari、build-in browsers of major apps.


But, they can not work in Android System, including built-in browsers of XiaoMi System, and various built-in browsers of major apps such as QQ、WeChat、Feishu.


I studied the source code (closebrackets.js)，find handleChar&nbsp;function doesn't work.


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "codemirror/CodeMirror"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年8月4日(星期二) 下午3:14
收件人:&nbsp;"codemirror/CodeMirror"<CodeMirror@noreply.github.com&gt;;
抄送:&nbsp;"阿轰不才"<1002363357@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [codemirror/CodeMirror] addon/closebrackets.js can not work in mobile (#6372)





 
It'd be nice if you could put a bit more effort into a bug report. Does 'mobile' mean Chrome Android in this case, or did you test more browsers?

**Comment by marijnh** [comment]
Right, Chrome Android doesn't fire key events, which is how closebrackets is responding to brackets, so indeed, this won't work. In the rewrite of the library (version 6) this should be better (there closebrackets observes actual user input, not key events), but I don't expect it will be fixed in version 5.

**Comment by ahongdream** [comment]
Thanks, I'm looking forward to version 6




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "codemirror/CodeMirror"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年8月11日(星期二) 下午4:46
收件人:&nbsp;"codemirror/CodeMirror"<CodeMirror@noreply.github.com&gt;;
抄送:&nbsp;"阿轰不才"<1002363357@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [codemirror/CodeMirror] addon/closebrackets.js can not work in mobile (#6372)





 
Right, Chrome Android doesn't fire key events, which is how closebrackets is responding to brackets, so indeed, this won't work. In the rewrite of the library (version 6) this should be better (there closebrackets observes actual user input, not key events), but I don't expect it will be fixed in version 5.
